In [81]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [82]:
data=pd.read_csv('train.csv')
data


Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type     Mileage  \
0                 41000    Diesel       Manual      First  19.67 kmpl   
1                 46000    Petrol       Manual      First    13 km/kg   
2                 87000    Diesel       Manual      First  20.77 kmpl   
3                 40670    Diesel    Automatic     Second   15.2 kmpl   
4                 86999    Diesel       Manual      First  23.08 kmpl   
...                 ...       ...          ...        ...         ...   
5842              27365    Diesel       Manual      First   28.4 kmpl   
5843             100000    Diesel       Manual      First   24.4 kmpl   
5844              55000    Diesel       Manual     Second   14.0 kmpl   
5845              46000    Petrol       Manual      First   18.9 kmpl   
5846              47000    Diesel       Manual      First  25.44 kmpl   

       Engine      Power  Seats  New_Price  Price  
0     1582 CC  126.2 bhp    5.0        NaN  12.50  
1     1199 CC   88.7 bhp    5.0  8.61 Lakh   4.50  
2     1248 CC  88.76 bhp    7.0        NaN   6.00  
3     1968 CC  140.8 bhp    5.0        NaN  17.74  
4     1461 CC   63.1 bhp    5.0        NaN   3.50  
...       ...        ...    ...        ...    ...  
5842  1248 CC     74 bhp    5.0  7.88 Lakh   4.75  
5843  1120 CC     71 bhp    5.0        NaN   4.00  
5844  2498 CC    112 bhp    8.0        NaN   2.90  
5845   998 CC   67.1 bhp    5.0        NaN   2.65  
5846   936 CC   57.6 bhp    5.0        NaN   2.50  

[5847 rows x 14 columns]

In [83]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5847 entries, 0 to 5846
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         5847 non-null   int64  
 1   Name               5847 non-null   object 
 2   Location           5847 non-null   object 
 3   Year               5847 non-null   int64  
 4   Kilometers_Driven  5847 non-null   int64  
 5   Fuel_Type          5847 non-null   object 
 6   Transmission       5847 non-null   object 
 7   Owner_Type         5847 non-null   object 
 8   Mileage            5845 non-null   object 
 9   Engine             5811 non-null   object 
 10  Power              5811 non-null   object 
 11  Seats              5809 non-null   float64
 12  New_Price          815 non-null    object 
 13  Price              5847 non-null   float64
dtypes: float64(2), int64(3), object(9)
memory usage: 639.6+ KB


In the 14 attributes there are 5 attributes with the null values

In [84]:
missing = data.isnull().sum()
missing

Unnamed: 0              0
Name                    0
Location                0
Year                    0
Kilometers_Driven       0
Fuel_Type               0
Transmission            0
Owner_Type              0
Mileage                 2
Engine                 36
Power                  36
Seats                  38
New_Price            5032
Price                   0
dtype: int64

In [85]:
#mileage median


Mileage is the attribute with the numerical type,so used median to fill the missing values ,which is effective instead of droping because there is small number of missing values i.e;2.
Firstly removed the units for the values in the mileage attribute then calculated median and filled the value in the missing place

In [86]:
data['Mileage'] = data['Mileage'].str.replace(r'kmpl', '', regex=True)
data['Mileage'] = data['Mileage'].str.replace(r'km/kg', '', regex=True)

In [87]:
mileage_median = data['Mileage'].median()
mileage_median

18.19

In [88]:
data['Mileage'].fillna(mileage_median, inplace=True)
data

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type Mileage   Engine  \
0                 41000    Diesel       Manual      First  19.67   1582 CC   
1                 46000    Petrol       Manual      First     13   1199 CC   
2                 87000    Diesel       Manual      First  20.77   1248 CC   
3                 40670    Diesel    Automatic     Second   15.2   1968 CC   
4                 86999    Diesel       Manual      First  23.08   1461 CC   
...                 ...       ...          ...        ...     ...      ...   
5842              27365    Diesel       Manual      First   28.4   1248 CC   
5843             100000    Diesel       Manual      First   24.4   1120 CC   
5844              55000    Diesel       Manual     Second   14.0   2498 CC   
5845              46000    Petrol       Manual      First   18.9    998 CC   
5846              47000    Diesel       Manual      First  25.44    936 CC   

          Power  Seats  New_Price  Price  
0     126.2 bhp    5.0        NaN  12.50  
1      88.7 bhp    5.0  8.61 Lakh   4.50  
2     88.76 bhp    7.0        NaN   6.00  
3     140.8 bhp    5.0        NaN  17.74  
4      63.1 bhp    5.0        NaN   3.50  
...         ...    ...        ...    ...  
5842     74 bhp    5.0  7.88 Lakh   4.75  
5843     71 bhp    5.0        NaN   4.00  
5844    112 bhp    8.0        NaN   2.90  
5845   67.1 bhp    5.0        NaN   2.65  
5846   57.6 bhp    5.0        NaN   2.50  

[5847 rows x 14 columns]

In [89]:
missing = data.isnull().sum()
missing

Unnamed: 0              0
Name                    0
Location                0
Year                    0
Kilometers_Driven       0
Fuel_Type               0
Transmission            0
Owner_Type              0
Mileage                 0
Engine                 36
Power                  36
Seats                  38
New_Price            5032
Price                   0
dtype: int64

In [90]:
#seats mode



seats is the attribute with the numerical type,so used mode to fill the missing values ,which is effective instead of mean and median because mode gives the perfect number where as mean, median gives float which is incompatable value for the car seats.

In [91]:
seats_mode = data['Seats'].mode()[0]
seats_mode

5.0

In [92]:
data['Seats'].fillna(seats_mode, inplace=True)
data

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type Mileage   Engine  \
0                 41000    Diesel       Manual      First  19.67   1582 CC   
1                 46000    Petrol       Manual      First     13   1199 CC   
2                 87000    Diesel       Manual      First  20.77   1248 CC   
3                 40670    Diesel    Automatic     Second   15.2   1968 CC   
4                 86999    Diesel       Manual      First  23.08   1461 CC   
...                 ...       ...          ...        ...     ...      ...   
5842              27365    Diesel       Manual      First   28.4   1248 CC   
5843             100000    Diesel       Manual      First   24.4   1120 CC   
5844              55000    Diesel       Manual     Second   14.0   2498 CC   
5845              46000    Petrol       Manual      First   18.9    998 CC   
5846              47000    Diesel       Manual      First  25.44    936 CC   

          Power  Seats  New_Price  Price  
0     126.2 bhp    5.0        NaN  12.50  
1      88.7 bhp    5.0  8.61 Lakh   4.50  
2     88.76 bhp    7.0        NaN   6.00  
3     140.8 bhp    5.0        NaN  17.74  
4      63.1 bhp    5.0        NaN   3.50  
...         ...    ...        ...    ...  
5842     74 bhp    5.0  7.88 Lakh   4.75  
5843     71 bhp    5.0        NaN   4.00  
5844    112 bhp    8.0        NaN   2.90  
5845   67.1 bhp    5.0        NaN   2.65  
5846   57.6 bhp    5.0        NaN   2.50  

[5847 rows x 14 columns]

In [93]:
missing = data.isnull().sum()
missing

Unnamed: 0              0
Name                    0
Location                0
Year                    0
Kilometers_Driven       0
Fuel_Type               0
Transmission            0
Owner_Type              0
Mileage                 0
Engine                 36
Power                  36
Seats                   0
New_Price            5032
Price                   0
dtype: int64

In [94]:
#power median


power is the attribute with the numerical type,so used median to fill the missing values ,which is effective instead of droping because there is small number of missing values i.e;36.firstly removed the units for the values in the power attribute then calculated median and filled the value in the missing place

In [95]:
data['Power'] = data['Power'].str.replace(r'bhp', '', regex=True)

In [96]:
power_median = data['Power'].median()
power_median

98.6

In [97]:
data['Power'].fillna(power_median, inplace=True)
data

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type Mileage   Engine  \
0                 41000    Diesel       Manual      First  19.67   1582 CC   
1                 46000    Petrol       Manual      First     13   1199 CC   
2                 87000    Diesel       Manual      First  20.77   1248 CC   
3                 40670    Diesel    Automatic     Second   15.2   1968 CC   
4                 86999    Diesel       Manual      First  23.08   1461 CC   
...                 ...       ...          ...        ...     ...      ...   
5842              27365    Diesel       Manual      First   28.4   1248 CC   
5843             100000    Diesel       Manual      First   24.4   1120 CC   
5844              55000    Diesel       Manual     Second   14.0   2498 CC   
5845              46000    Petrol       Manual      First   18.9    998 CC   
5846              47000    Diesel       Manual      First  25.44    936 CC   

       Power  Seats  New_Price  Price  
0     126.2     5.0        NaN  12.50  
1      88.7     5.0  8.61 Lakh   4.50  
2     88.76     7.0        NaN   6.00  
3     140.8     5.0        NaN  17.74  
4      63.1     5.0        NaN   3.50  
...      ...    ...        ...    ...  
5842     74     5.0  7.88 Lakh   4.75  
5843     71     5.0        NaN   4.00  
5844    112     8.0        NaN   2.90  
5845   67.1     5.0        NaN   2.65  
5846   57.6     5.0        NaN   2.50  

[5847 rows x 14 columns]

In [98]:
missing = data.isnull().sum()
missing

Unnamed: 0              0
Name                    0
Location                0
Year                    0
Kilometers_Driven       0
Fuel_Type               0
Transmission            0
Owner_Type              0
Mileage                 0
Engine                 36
Power                   0
Seats                   0
New_Price            5032
Price                   0
dtype: int64

In [99]:
#engine mean

engine is the attribute with the numerical type,so used mean to fill the missing values ,which is effective instead of droping because there is small number of missing values i.e;36.firstly removed the units for the values in the engine attribute then calculated mean value and filled the value in the missing place

In [100]:
data['Engine'] = data['Engine'].str.replace(r'CC', '', regex=True)

In [101]:
data['Engine'] = pd.to_numeric(data['Engine'], errors='coerce')

In [102]:
engine_mean = data['Engine'].mean()
engine_mean

1631.552572706935

In [103]:
engine_mean = int(engine_mean)
engine_mean

1631

In [104]:
data['Engine'].fillna(engine_mean, inplace=True)
data

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type Mileage  Engine  \
0                 41000    Diesel       Manual      First  19.67   1582.0   
1                 46000    Petrol       Manual      First     13   1199.0   
2                 87000    Diesel       Manual      First  20.77   1248.0   
3                 40670    Diesel    Automatic     Second   15.2   1968.0   
4                 86999    Diesel       Manual      First  23.08   1461.0   
...                 ...       ...          ...        ...     ...     ...   
5842              27365    Diesel       Manual      First   28.4   1248.0   
5843             100000    Diesel       Manual      First   24.4   1120.0   
5844              55000    Diesel       Manual     Second   14.0   2498.0   
5845              46000    Petrol       Manual      First   18.9    998.0   
5846              47000    Diesel       Manual      First  25.44    936.0   

       Power  Seats  New_Price  Price  
0     126.2     5.0        NaN  12.50  
1      88.7     5.0  8.61 Lakh   4.50  
2     88.76     7.0        NaN   6.00  
3     140.8     5.0        NaN  17.74  
4      63.1     5.0        NaN   3.50  
...      ...    ...        ...    ...  
5842     74     5.0  7.88 Lakh   4.75  
5843     71     5.0        NaN   4.00  
5844    112     8.0        NaN   2.90  
5845   67.1     5.0        NaN   2.65  
5846   57.6     5.0        NaN   2.50  

[5847 rows x 14 columns]

In [105]:
missing = data.isnull().sum()
missing

Unnamed: 0              0
Name                    0
Location                0
Year                    0
Kilometers_Driven       0
Fuel_Type               0
Transmission            0
Owner_Type              0
Mileage                 0
Engine                  0
Power                   0
Seats                   0
New_Price            5032
Price                   0
dtype: int64

In [106]:
#new_price dropping

New_price is the attribute with the numerical type,Used dropping the rows because there are 5032 null values,which is effective instead of filling the missing values with the mean,median,mode.firstly removed the units for the values in the new_price attribute then dropped the rows with the missing values.

In [107]:
data['New_Price']=data['New_Price'].str.extract('(\d+\.\d+)').astype(float)

In [108]:
d1=data.dropna()
d1

Unnamed: 0                           Name Location  Year  \
1              2                   Honda Jazz V  Chennai  2011   
8             10               Maruti Ciaz Zeta    Kochi  2018   
13            15    Mitsubishi Pajero Sport 4X4    Delhi  2014   
18            20              BMW 3 Series 320d    Kochi  2014   
26            28            Honda WRV i-VTEC VX    Kochi  2018   
...          ...                            ...      ...   ...   
5827        5999          Tata Bolt Revotron XT  Chennai  2016   
5830        6002  Volkswagen Vento 1.6 Highline   Mumbai  2011   
5833        6005       Maruti Vitara Brezza VDi     Pune  2016   
5838        6010           Honda Brio 1.2 VX MT    Delhi  2013   
5842        6014               Maruti Swift VDI    Delhi  2014   

      Kilometers_Driven Fuel_Type Transmission Owner_Type Mileage  Engine  \
1                 46000    Petrol       Manual      First     13   1199.0   
8                 25692    Petrol       Manual      First  21.56   1462.0   
13               110000    Diesel       Manual      First   13.5   2477.0   
18                32982    Diesel    Automatic      First  22.69   1995.0   
26                37430    Petrol       Manual      First   17.5   1199.0   
...                 ...       ...          ...        ...     ...     ...   
5827              10000    Petrol       Manual      First  17.57   1193.0   
5830              38000    Petrol       Manual      First  16.09   1598.0   
5833              37208    Diesel       Manual      First   24.3   1248.0   
5838              33746    Petrol       Manual      First   18.5   1198.0   
5842              27365    Diesel       Manual      First   28.4   1248.0   

        Power  Seats  New_Price  Price  
1       88.7     5.0       8.61   4.50  
8     103.25     5.0      10.65   9.95  
13    175.56     7.0      32.01  15.00  
18       190     5.0      47.87  18.55  
26      88.7     5.0      10.57   9.90  
...       ...    ...        ...    ...  
5827    88.7     5.0       7.77   4.00  
5830   103.5     5.0      11.91   3.25  
5833    88.5     5.0       9.93   7.43  
5838    86.8     5.0       6.63   3.20  
5842      74     5.0       7.88   4.75  

[801 rows x 14 columns]

In [109]:
missing = d1.isnull().sum()
missing

Unnamed: 0           0
Name                 0
Location             0
Year                 0
Kilometers_Driven    0
Fuel_Type            0
Transmission         0
Owner_Type           0
Mileage              0
Engine               0
Power                0
Seats                0
New_Price            0
Price                0
dtype: int64

c)Changed the categorical values in to numerical values like Diesel,petrol,electric-0,1,2 respectively,maual as 1 automatic as 2.

In [110]:
d1['Fuel_Type'].replace(['Diesel', 'Petrol','Electric'], [0,1,2],inplace=True)
d1['Transmission'].replace(['Manual', 'Automatic'], [1, 2],inplace=True)
d1['Owner_Type'].replace(['First', 'Second','Third','Fourth & Above'], [1,2,3,4],inplace=True)
d1

<ipython-input-110-b0155440198e>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d1['Fuel_Type'].replace(['Diesel', 'Petrol','Electric'], [0,1,2],inplace=True)
<ipython-input-110-b0155440198e>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d1['Transmission'].replace(['Manual', 'Automatic'], [1, 2],inplace=True)
<ipython-input-110-b0155440198e>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d1['Owner_Type'].replace(['First', 'Secon

Unnamed: 0                           Name Location  Year  \
1              2                   Honda Jazz V  Chennai  2011   
8             10               Maruti Ciaz Zeta    Kochi  2018   
13            15    Mitsubishi Pajero Sport 4X4    Delhi  2014   
18            20              BMW 3 Series 320d    Kochi  2014   
26            28            Honda WRV i-VTEC VX    Kochi  2018   
...          ...                            ...      ...   ...   
5827        5999          Tata Bolt Revotron XT  Chennai  2016   
5830        6002  Volkswagen Vento 1.6 Highline   Mumbai  2011   
5833        6005       Maruti Vitara Brezza VDi     Pune  2016   
5838        6010           Honda Brio 1.2 VX MT    Delhi  2013   
5842        6014               Maruti Swift VDI    Delhi  2014   

      Kilometers_Driven  Fuel_Type  Transmission  Owner_Type Mileage  Engine  \
1                 46000          1             1           1     13   1199.0   
8                 25692          1             1           1  21.56   1462.0   
13               110000          0             1           1   13.5   2477.0   
18                32982          0             2           1  22.69   1995.0   
26                37430          1             1           1   17.5   1199.0   
...                 ...        ...           ...         ...     ...     ...   
5827              10000          1             1           1  17.57   1193.0   
5830              38000          1             1           1  16.09   1598.0   
5833              37208          0             1           1   24.3   1248.0   
5838              33746          1             1           1   18.5   1198.0   
5842              27365          0             1           1   28.4   1248.0   

        Power  Seats  New_Price  Price  
1       88.7     5.0       8.61   4.50  
8     103.25     5.0      10.65   9.95  
13    175.56     7.0      32.01  15.00  
18       190     5.0      47.87  18.55  
26      88.7     5.0      10.57   9.90  
...       ...    ...        ...    ...  
5827    88.7     5.0       7.77   4.00  
5830   103.5     5.0      11.91   3.25  
5833    88.5     5.0       9.93   7.43  
5838    86.8     5.0       6.63   3.20  
5842      74     5.0       7.88   4.75  

[801 rows x 14 columns]

d)created the curreent_age attribute by substracting with the currect year and year from the data set

In [111]:
from datetime import datetime
current_year=datetime.now().year
d1['current_age']=current_year-d1['Year']
d1

<ipython-input-111-33d30f71e61b>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d1['current_age']=current_year-d1['Year']


Unnamed: 0                           Name Location  Year  \
1              2                   Honda Jazz V  Chennai  2011   
8             10               Maruti Ciaz Zeta    Kochi  2018   
13            15    Mitsubishi Pajero Sport 4X4    Delhi  2014   
18            20              BMW 3 Series 320d    Kochi  2014   
26            28            Honda WRV i-VTEC VX    Kochi  2018   
...          ...                            ...      ...   ...   
5827        5999          Tata Bolt Revotron XT  Chennai  2016   
5830        6002  Volkswagen Vento 1.6 Highline   Mumbai  2011   
5833        6005       Maruti Vitara Brezza VDi     Pune  2016   
5838        6010           Honda Brio 1.2 VX MT    Delhi  2013   
5842        6014               Maruti Swift VDI    Delhi  2014   

      Kilometers_Driven  Fuel_Type  Transmission  Owner_Type Mileage  Engine  \
1                 46000          1             1           1     13   1199.0   
8                 25692          1             1           1  21.56   1462.0   
13               110000          0             1           1   13.5   2477.0   
18                32982          0             2           1  22.69   1995.0   
26                37430          1             1           1   17.5   1199.0   
...                 ...        ...           ...         ...     ...     ...   
5827              10000          1             1           1  17.57   1193.0   
5830              38000          1             1           1  16.09   1598.0   
5833              37208          0             1           1   24.3   1248.0   
5838              33746          1             1           1   18.5   1198.0   
5842              27365          0             1           1   28.4   1248.0   

        Power  Seats  New_Price  Price  current_age  
1       88.7     5.0       8.61   4.50           13  
8     103.25     5.0      10.65   9.95            6  
13    175.56     7.0      32.01  15.00           10  
18       190     5.0      47.87  18.55           10  
26      88.7     5.0      10.57   9.90            6  
...       ...    ...        ...    ...          ...  
5827    88.7     5.0       7.77   4.00            8  
5830   103.5     5.0      11.91   3.25           13  
5833    88.5     5.0       9.93   7.43            8  
5838    86.8     5.0       6.63   3.20           11  
5842      74     5.0       7.88   4.75           10  

[801 rows x 15 columns]

Performing select, filter, rename, mutate, arrange and summarize with group by operations on this dataset

SELECT is the operation where it selects few attributes according to our choice

In [112]:
selection=d1[['Name','Seats','New_Price','Fuel_Type']]
selection

Name  Seats  New_Price  Fuel_Type
1                      Honda Jazz V    5.0       8.61          1
8                  Maruti Ciaz Zeta    5.0      10.65          1
13      Mitsubishi Pajero Sport 4X4    7.0      32.01          0
18                BMW 3 Series 320d    5.0      47.87          0
26              Honda WRV i-VTEC VX    5.0      10.57          1
...                             ...    ...        ...        ...
5827          Tata Bolt Revotron XT    5.0       7.77          1
5830  Volkswagen Vento 1.6 Highline    5.0      11.91          1
5833       Maruti Vitara Brezza VDi    5.0       9.93          0
5838           Honda Brio 1.2 VX MT    5.0       6.63          1
5842               Maruti Swift VDI    5.0       7.88          0

[801 rows x 4 columns]

Filter is just to filter the dataset with the specific value

In [113]:
filter=d1[d1['New_Price']>10]
filter

Unnamed: 0                                 Name    Location  Year  \
8             10                     Maruti Ciaz Zeta       Kochi  2018   
13            15          Mitsubishi Pajero Sport 4X4       Delhi  2014   
18            20                    BMW 3 Series 320d       Kochi  2014   
26            28                  Honda WRV i-VTEC VX       Kochi  2018   
32            34        Volkswagen Vento 1.6 Highline  Coimbatore  2014   
...          ...                                  ...         ...   ...   
5794        5964  Hyundai Verna VTVT 1.6 AT SX Option      Mumbai  2018   
5800        5970    Mercedes-Benz GLA Class 200 Sport       Kochi  2018   
5801        5971     Volkswagen Polo 1.5 TDI Highline   Hyderabad  2013   
5818        5989          Audi A4 35 TDI Premium Plus  Coimbatore  2013   
5830        6002        Volkswagen Vento 1.6 Highline      Mumbai  2011   

      Kilometers_Driven  Fuel_Type  Transmission  Owner_Type Mileage  Engine  \
8                 25692          1             1           1  21.56   1462.0   
13               110000          0             1           1   13.5   2477.0   
18                32982          0             2           1  22.69   1995.0   
26                37430          1             1           1   17.5   1199.0   
32                55431          1             1           2  16.09   1598.0   
...                 ...        ...           ...         ...     ...     ...   
5794               8000          1             2           1   17.0   1591.0   
5800              17773          1             2           1   13.7   1991.0   
5801              65000          0             1           2  20.14   1498.0   
5818              58629          0             2           2  18.25   1968.0   
5830              38000          1             1           1  16.09   1598.0   

        Power  Seats  New_Price  Price  current_age  
8     103.25     5.0      10.65   9.95            6  
13    175.56     7.0      32.01  15.00           10  
18       190     5.0      47.87  18.55           10  
26      88.7     5.0      10.57   9.90            6  
32     103.5     5.0      12.33   6.98           10  
...       ...    ...        ...    ...          ...  
5794   121.3     5.0      15.14  12.85            6  
5800     183     5.0      39.22  26.76            6  
5801      88     5.0      10.15   3.50           11  
5818  187.74     5.0      53.14  16.52           11  
5830   103.5     5.0      11.91   3.25           13  

[485 rows x 15 columns]

Rename is to alter the attribute name

In [114]:
renamed = d1.rename(columns={'Kilometers_Driven':'Km_driven'})
renamed

Unnamed: 0                           Name Location  Year  Km_driven  \
1              2                   Honda Jazz V  Chennai  2011      46000   
8             10               Maruti Ciaz Zeta    Kochi  2018      25692   
13            15    Mitsubishi Pajero Sport 4X4    Delhi  2014     110000   
18            20              BMW 3 Series 320d    Kochi  2014      32982   
26            28            Honda WRV i-VTEC VX    Kochi  2018      37430   
...          ...                            ...      ...   ...        ...   
5827        5999          Tata Bolt Revotron XT  Chennai  2016      10000   
5830        6002  Volkswagen Vento 1.6 Highline   Mumbai  2011      38000   
5833        6005       Maruti Vitara Brezza VDi     Pune  2016      37208   
5838        6010           Honda Brio 1.2 VX MT    Delhi  2013      33746   
5842        6014               Maruti Swift VDI    Delhi  2014      27365   

      Fuel_Type  Transmission  Owner_Type Mileage  Engine    Power  Seats  \
1             1             1           1     13   1199.0    88.7     5.0   
8             1             1           1  21.56   1462.0  103.25     5.0   
13            0             1           1   13.5   2477.0  175.56     7.0   
18            0             2           1  22.69   1995.0     190     5.0   
26            1             1           1   17.5   1199.0    88.7     5.0   
...         ...           ...         ...     ...     ...      ...    ...   
5827          1             1           1  17.57   1193.0    88.7     5.0   
5830          1             1           1  16.09   1598.0   103.5     5.0   
5833          0             1           1   24.3   1248.0    88.5     5.0   
5838          1             1           1   18.5   1198.0    86.8     5.0   
5842          0             1           1   28.4   1248.0      74     5.0   

      New_Price  Price  current_age  
1          8.61   4.50           13  
8         10.65   9.95            6  
13        32.01  15.00           10  
18        47.87  18.55           10  
26        10.57   9.90            6  
...         ...    ...          ...  
5827       7.77   4.00            8  
5830      11.91   3.25           13  
5833       9.93   7.43            8  
5838       6.63   3.20           11  
5842       7.88   4.75           10  

[801 rows x 15 columns]

mutate is create the attribute or changing the existing atttribute values.

In [115]:
d1['price_difference'] = d1['New_Price'] - d1['Price']
d1

<ipython-input-115-9d157a3de565>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d1['price_difference'] = d1['New_Price'] - d1['Price']


Unnamed: 0                           Name Location  Year  \
1              2                   Honda Jazz V  Chennai  2011   
8             10               Maruti Ciaz Zeta    Kochi  2018   
13            15    Mitsubishi Pajero Sport 4X4    Delhi  2014   
18            20              BMW 3 Series 320d    Kochi  2014   
26            28            Honda WRV i-VTEC VX    Kochi  2018   
...          ...                            ...      ...   ...   
5827        5999          Tata Bolt Revotron XT  Chennai  2016   
5830        6002  Volkswagen Vento 1.6 Highline   Mumbai  2011   
5833        6005       Maruti Vitara Brezza VDi     Pune  2016   
5838        6010           Honda Brio 1.2 VX MT    Delhi  2013   
5842        6014               Maruti Swift VDI    Delhi  2014   

      Kilometers_Driven  Fuel_Type  Transmission  Owner_Type Mileage  Engine  \
1                 46000          1             1           1     13   1199.0   
8                 25692          1             1           1  21.56   1462.0   
13               110000          0             1           1   13.5   2477.0   
18                32982          0             2           1  22.69   1995.0   
26                37430          1             1           1   17.5   1199.0   
...                 ...        ...           ...         ...     ...     ...   
5827              10000          1             1           1  17.57   1193.0   
5830              38000          1             1           1  16.09   1598.0   
5833              37208          0             1           1   24.3   1248.0   
5838              33746          1             1           1   18.5   1198.0   
5842              27365          0             1           1   28.4   1248.0   

        Power  Seats  New_Price  Price  current_age  price_difference  
1       88.7     5.0       8.61   4.50           13              4.11  
8     103.25     5.0      10.65   9.95            6              0.70  
13    175.56     7.0      32.01  15.00           10             17.01  
18       190     5.0      47.87  18.55           10             29.32  
26      88.7     5.0      10.57   9.90            6              0.67  
...       ...    ...        ...    ...          ...               ...  
5827    88.7     5.0       7.77   4.00            8              3.77  
5830   103.5     5.0      11.91   3.25           13              8.66  
5833    88.5     5.0       9.93   7.43            8              2.50  
5838    86.8     5.0       6.63   3.20           11              3.43  
5842      74     5.0       7.88   4.75           10              3.13  

[801 rows x 16 columns]

Arranged is to arrange the values in the particular attribute

In [116]:
arranged = d1.sort_values(by='Seats')
arranged

Unnamed: 0                               Name    Location  Year  \
2035        2095           Mercedes-Benz SLC 43 AMG  Coimbatore  2019   
127          134           Mercedes-Benz SLC 43 AMG     Kolkata  2017   
2294        2364          Mini Cooper Convertible S     Chennai  2012   
411          427          Mini Cooper Convertible S      Mumbai  2014   
4318        4451                    Ford Mustang V8       Kochi  2018   
...          ...                                ...         ...   ...   
2847        2937  Toyota Innova Crysta 2.8 GX AT 8S       Delhi  2017   
1484        1532  Toyota Innova Crysta 2.4 VX MT 8S  Coimbatore  2019   
3783        3907  Toyota Innova Crysta 2.8 GX AT 8S     Chennai  2016   
2674        2763  Toyota Innova Crysta 2.4 VX MT 8S  Coimbatore  2017   
58            60  Toyota Innova Crysta 2.4 GX MT 8S      Mumbai  2017   

      Kilometers_Driven  Fuel_Type  Transmission  Owner_Type Mileage  Engine  \
2035               2526          1             2           1   19.0   2996.0   
127               13372          1             2           1   19.0   2996.0   
2294              17000          1             2           2  16.82   1998.0   
411               11900          1             2           1  16.82   1998.0   
4318              21290          1             2           1   13.0   4951.0   
...                 ...        ...           ...         ...     ...     ...   
2847              18000          0             2           1  11.36   2755.0   
1484              11241          0             1           1  13.68   2393.0   
3783              40000          0             2           1  11.36   2755.0   
2674              72202          0             1           1  13.68   2393.0   
58                22000          0             1           1  13.68   2393.0   

        Power  Seats  New_Price  Price  current_age  price_difference  
2035  362.07     2.0       1.06  83.96            5            -82.90  
127   362.07     2.0      95.04  54.00            7             41.04  
2294  189.08     4.0      44.28  22.00           12             22.28  
411   189.08     4.0      44.28  28.50           10             15.78  
4318     395     4.0      83.92  56.80            6             27.12  
...       ...    ...        ...    ...          ...               ...  
2847   171.5     8.0      20.80  18.25            7              2.55  
1484   147.8     8.0      24.01  22.55            5              1.46  
3783   171.5     8.0      21.10  16.45            8              4.65  
2674   147.8     8.0      24.01  17.92            7              6.09  
58     147.8     8.0      19.34  16.50            7              2.84  

[801 rows x 16 columns]

group by operation , here i just checked the mean price for petrol,diesel and electirc

In [117]:
summary = d1.groupby('Fuel_Type')['Price'].mean()
summary

Fuel_Type
0    18.081856
1     8.600786
2    13.000000
Name: Price, dtype: float64